src/sentry/middleware/devtoolbar.py

```diff
@@ -0,0 +1,64 @@
+import logging
+
+from sentry import analytics
+from sentry.utils.http import (
+    get_api_path_from_request,
+    origin_from_request,
+    parse_id_or_slug_param,
+    query_string,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class AnalyticsMiddleware:
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request):
+        return self.get_response(request)
+
+    def process_view(self, request, _view_func, _view_args, view_kwargs) -> None:
+        # Intercepts requests right before they're passed to the Endpoint/View.
+        org_id, org_slug, project_id, project_slug = None, None, None, None
+        try:
+            if request.headers.get("queryReferrer") == "devtoolbar":
```

Comment:
prefer return early for readability https://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html